### PR TITLE
Add CDK stacks for steps 1, 2, 6

### PR DESCRIPTION
**Pull request by seeebiii**
Hey @HenrikFricke & @tillkahlbrock,

I have participated in your AppSync/GraphQL workshop on Thursday and really enjoyed it! One participant of the workshop also asked for an infrastructure as code example. Since I have participated in the CDK workshop on Thursday afternoon, I thought providing the GraphQL examples as CDK stacks is a good practice for me 😄 

I have only provided examples for steps 1, 2, and 6 since these require most of the code. The other steps only require a few minor GraphQL API settings but I can provide them as well if you like.

Also, you'll see that there's a lot of duplicated code for the stack classes of the different steps. I haven't created a CDK construct or similar because I thought the stacks might be easier to understand if someone isn't really familiar with GraphQL and CDK as well but has at least some basic understanding of infrastructure as code.

Let me know any feedback as I'm not super familiar with CDK yet but I've really enjoyed working with it. And if you don't need this PR for your workshop, also fine :)

### CODE REVIEW

File: infrastructure/lib/infrastructure-stack1.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    });
+
+    new appSync.CfnApiKey(this, 'BlogApiKey', {
+      apiId: blogsGraphQLApi.attrApiId
+    });
+
+
+    /////////// GRAPHQL API SCHEMA ///////////
+
+    const apiSchema = new appSync.CfnGraphQLSchema(this, 'BlogSchema', {
```
Comment (HenrikFricke) [inline infrastructure/lib/infrastructure-stack1.ts]: Tip: Quite often CDK tries to abstract the Cloudformation constructs by introducing new ones. The benefit is that we have to write less boilerplate code, e.g. dealing with roles and policies. In this case, there is a `GraphqlApi` class, so we don't have to use `CfnGraphQLApi`, `CfnApiKey`, `CfnGraphQLSchema`. 

See: https://docs.aws.amazon.com/cdk/api/latest/docs/aws-appsync-readme.html#dynamodb

File: infrastructure/lib/infrastructure-stack1.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      apiId: blogsGraphQLApi.attrApiId,
+      definition: `
```
Comment (HenrikFricke) [inline infrastructure/lib/infrastructure-stack1.ts]: Tip: You can move the schema also in a dedicated file to keep the stack cleaner. CDK provides a `Schema.fromAsset` helper. 

https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-appsync.Schema.html#static-from-wbr-assetfilepath-span-class-api-icon-api-icon-experimental-title-this-api-element-is-experimental-it-may-change-without-notice-span

File: infrastructure/lib/infrastructure-stack2.ts [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    });
+    commentsDataSource.addDependsOn(apiSchema);
+
+
+    /////////// GRAPHQL DATA SOURCE - BAD EMOJI LAMBDA FUNCTION ///////////
+
+    const hasBadEmojisLambda = new Function(this, 'hasBadEmojisLambda', {
+      runtime: Runtime.NODEJS_12_X,
+      code: Code.fromAsset('lambda'),
+      handler: 'commentValidator.handler'
```
Comment (HenrikFricke) [inline infrastructure/lib/infrastructure-stack2.ts]: Hm, I can't find this handler 🤔

**Review by HenrikFricke — COMMENTED** [review]
Wow, I'm really impressed! Thank you so much for contributing to our workshop! 🥳

I added some comments: As a tip, when working with CDK we should try to avoid the `Cfn…` classes as much as we can and search for alternatives. Quite often CDK provides good helpers to abstract the whole IAM, policy, role handling, etc. so we have to write less code and also it's easier for us to navigate through the stacks. 

Other than that, I like the idea of having an infrastructure folder and stacks per lab so we can always deploy the labs individually.

Thanks again 🙏

**Comment by seeebiii** [comment]
Thanks for your review and hints! I'll update the code within the next few days or on the weekend, so you can have another look! :)

File: infrastructure/lib/infrastructure-stack2.ts (hunk 3 above)
Comment (seeebiii) [reply infrastructure/lib/infrastructure-stack2.ts]: Oops, I'll check that again 😄

**Comment by seeebiii** [comment]
Hey @HenrikFricke , I've updated the code. Thanks for all the hints, they were really helpful! However, I couldn't find a replacement for `CfnFunctionConfiguration` - if you have a suggestion, just let me know :)

**Comment by HenrikFricke** [comment]
> Hey @HenrikFricke , I've updated the code. Thanks for all the hints, they were really helpful! However, I couldn't find a replacement for `CfnFunctionConfiguration` - if you have a suggestion, just let me know :)

Afaik CDK doesn't have anything in place for pipeline functions. I really like your refactoring! I'd like to merge it :)

**Pull request merged**